databricks/koalas/frame.py

```diff
@@ -2042,6 +2042,204 @@ class  locomotion
+        >>> df2 = ks.DataFrame({'A': [0, -1, -2, -3, -4], 'B':[-100, -200, -300, -400, -500]})
+        >>> df1
+           A    B
+        0  0  100
+        1  1  200
+        2  2  300
+        3  3  400
+        4  4  500
+        >>> df2
+           A    B
+        0  0 -100
+        1 -1 -200
+        2 -2 -300
+        3 -3 -400
+        4 -4 -500
+
+        >>> df1.where(df1 > 0).sort_index()
+             A      B
+        0  NaN  100.0
+        1  1.0  200.0
+        2  2.0  300.0
+        3  3.0  400.0
+        4  4.0  500.0
+
+        >>> df1.where(df1 > 1, 10).sort_index()
+            A    B
+        0  10  100
+        1  10  200
+        2   2  300
+        3   3  400
+        4   4  500
+
+        >>> df1.where(df1 > 1, df1 + 100).sort_index()
+             A    B
+        0  100  100
+        1  101  200
+        2    2  300
+        3    3  400
+        4    4  500
+
+        >>> df1.where(df1 > 1, df2).sort_index()
+           A    B
+        0  0  100
+        1 -1  200
+        2  2  300
+        3  3  400
+        4  4  500
+
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        tmp_cond_col_name = '__tmp_cond_col_{}__'
+        tmp_other_col_name = '__tmp_other_col_{}__'
+        kdf = self.copy()
+        for column in self._internal.data_columns:
+            kdf[tmp_cond_col_name.format(column)] = cond[column]
+            if isinstance(other, DataFrame):
+                kdf[tmp_other_col_name.format(column)] = other[column]
+            else:
+                kdf[tmp_other_col_name.format(column)] = other
+
+        # above logic make spark dataframe looks like below:
+        # +-----------------+---+---+------------------+-------------------+------------------+--...
+        # |__index_level_0__|  A|  B|__tmp_cond_col_A__|__tmp_other_col_A__|__tmp_cond_col_B__|__...
+        # +-----------------+---+---+------------------+-------------------+------------------+--...
+        # |                0|  0|100|              true|                  0|             false|  ...
+        # |                1|  1|200|             false|                 -1|             false|  ...
+        # |                3|  3|400|              true|                 -3|             false|  ...
+        # |                2|  2|300|             false|                 -2|              true|  ...
+        # |                4|  4|500|             false|                 -4|             false|  ...
+        # +-----------------+---+---+------------------+-------------------+------------------+--...
+
+        sdf = kdf._sdf
+        conditions = []
+        for column in self._internal.data_columns:
+            data_col_name = self._internal.column_name_for(column)
+            conditions.append(
+                F.when(sdf[tmp_cond_col_name.format(column)], sdf[data_col_name])
+                 .otherwise(sdf[tmp_other_col_name.format(column)]).alias(data_col_name))
```

Comment:
I would fix the indentation as below to make it easier to read.

```python
conditions.append(
    F.when(
        sdf[tmp_cond_col_name.format(column)], sdf[data_col_name]
    ).otherwise(
        sdf[tmp_other_col_name.format(column)]
    ).alias(data_col_name))
```

Reply:
looks way better, thanks!! 😄